TeaLeaf/jni/platform/native_shim.cpp

```diff
@@ -249,6 +249,14 @@ extern "C" {
         texture_manager_clear_textures(texture_manager_get(), true);
     }
 
+    void Java_com_tealeaf_NativeShim_textureManagerMemoryWarning(JNIEnv *env, jobject thiz) {
+        texture_manager_memory_warning();
+    }
+
+    void Java_com_tealeaf_NativeShim_textureManagerMemoryCritical(JNIEnv *env, jobject thiz) {
+        texture_manager_memory_critical();
```

Comment:
This is a new function in native-core. After we get a TRIM_MEMORY_RUNNING_CRITICAL  warning, we should not request more memory from the system, requesting will kill the app.